bot.ml

```diff
@@ -786,15 +786,158 @@ let callback _conn req body =
                   ()
           else if
             string_match ~regexp:(f "@%s: [Mm]erge now" bot_name) body
-            && Option.is_some comment_info.pull_request
-          then
-            (* Should be restricted to coq *)
+            && comment_info.issue.pull_request
+            && String.equal comment_info.issue.issue.owner "coq"
+            && String.equal comment_info.issue.issue.repo "coq"
+          then (
+            (fun () ->
+              let pr = comment_info.issue in
+              GitHub_queries.get_pull_request_reviews_refs ~bot_info
+                ~owner:pr.issue.owner ~repo:pr.issue.repo
+                ~number:pr.issue.number
+              >>= function
+              | Ok reviews_info -> (
+                match
+                  List.find comment_info.issue.labels ~f:(fun label ->
+                      string_match ~regexp:"needs:.*" label)
+                with
+                | Some l ->
+                    GitHub_mutations.post_comment ~bot_info
+                      ~message:
+                        (f "@%s: Needs label %s found, merge aborted."
+                           comment_info.author l)
+                      ~id:pr.id
+                | None -> (
+                    if
+                      not
+                        (List.exists comment_info.issue.labels ~f:(fun label ->
+                             string_match ~regexp:"kind:.*" label))
+                    then
+                      GitHub_mutations.post_comment ~bot_info
+                        ~message:
+                          (f "@%s: No kind label found, merge aborted."
+                             comment_info.author)
+                        ~id:pr.id
+                    else if not comment_info.issue.milestoned then
+                      GitHub_mutations.post_comment ~bot_info
+                        ~message:
+                          (f "@%s: No milestone found, merge aborted."
+                             comment_info.author)
+                        ~id:pr.id
+                    else
+                      match reviews_info.review_decision with
+                      | REVIEW_REQUIRED | CHANGES_REQUESTED ->
```

Comment:
In the case of `CHANGES_REQUESTED`, we might want to put a different message (as it might have been approved by some reviewers, but the problem is that some change requests remain).